### ISSUE TITLE
Error in runner.py when on "Load Pod an Container informations"

### ISSUE THREAD

**Issue by hugodopradofernandes** [klustair/klustair#3]
I sucessfully deployed using Helm chart, then I added the kube config file to klustair-configs secret.
When I try to run the cronjob using:
kubectl create job --from=cronjob/klustair-job klustair-job-manual-001 -n klustair-scan 

I receive an error:

```
klustair-runnerjob - DEBUG:Namespace: default
klustair-runnerjob - DEBUG:Namespace: ingress-nginx
klustair-runnerjob - DEBUG:Namespace: klustair-scan
klustair-runnerjob - DEBUG:Namespace: kube-node-lease
klustair-runnerjob - DEBUG:Namespace: kube-public
klustair-runnerjob - DEBUG:Namespace: monitoring
klustair-runnerjob - DEBUG:Namespace: nfs-client
klustair-runnerjob - DEBUG:Namespace: weave-scope
klustair-runnerjob - DEBUG:Namespace: wordpress
klustair-runnerjob - DEBUG:Kubeaudit: audit all on default
klustair-runnerjob - DEBUG:Kubeaudit: audit all on ingress-nginx
klustair-runnerjob - DEBUG:Kubeaudit: audit all on klustair-scan
klustair-runnerjob - DEBUG:Kubeaudit: audit all on kube-node-lease
klustair-runnerjob - DEBUG:Kubeaudit: audit all on kube-public
klustair-runnerjob - DEBUG:Kubeaudit: audit all on monitoring
klustair-runnerjob - DEBUG:Kubeaudit: audit all on nfs-client
klustair-runnerjob - DEBUG:Kubeaudit: audit all on weave-scope
klustair-runnerjob - DEBUG:Kubeaudit: audit all on wordpress
klustair-runnerjob - DEBUG:Pod: default-http-backend-598b7d7dbd-h2p7p
klustair-runnerjob - DEBUG:Container: default-http-backend
klustair-runnerjob - DEBUG:Pod: default-http-backend-598b7d7dbd-l5wld
klustair-runnerjob - DEBUG:Container: default-http-backend
klustair-runnerjob - INFO: Load Namespaces
klustair-runnerjob - INFO: Run Kubeaudit
klustair-runnerjob - not categorized: NamespaceHostNetworkTrue
klustair-runnerjob - not categorized: SeccompAnnotationMissing
klustair-runnerjob - not categorized: AutomountServiceAccountTokenTrueAndDefaultSA
klustair-runnerjob - not categorized: SeccompAnnotationMissing
klustair-runnerjob - not categorized: NamespaceHostNetworkTrue
klustair-runnerjob - not categorized: NamespaceHostPIDTrue
klustair-runnerjob - not categorized: SeccompAnnotationMissing
klustair-runnerjob - not categorized: SeccompAnnotationMissing
klustair-runnerjob - not categorized: AutomountServiceAccountTokenTrueAndDefaultSA
klustair-runnerjob - not categorized: NamespaceHostNetworkTrue
klustair-runnerjob - not categorized: NamespaceHostPIDTrue
klustair-runnerjob - not categorized: SeccompAnnotationMissing
klustair-runnerjob - INFO: Load Pod an Container informations
klustair-runnerjob - Traceback (most recent call last):
klustair-runnerjob - File "./runner.py", line 839, in
klustair-runnerjob - run()
klustair-runnerjob - File "./runner.py", line 768, in run
klustair-runnerjob - [podsList, containersList] = getPods(nsList)
klustair-runnerjob - File "./runner.py", line 154, in getPods
klustair-runnerjob - for containerStatus in pod['status']['containerStatuses']:
klustair-runnerjob - KeyError: 'containerStatuses'
```

Let me know if more details are required.

**Comment by mms-gianni** [comment]
I appriciate your help. Thank you. 

I never ran into this error. 

Which kubernetes version are you running on you cluster?

**Comment by hugodopradofernandes** [comment]
-----------------------------------------------------
Kubernetes v1.19.3

```
[rke]$ kubectl version 
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:50:19Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:41:49Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
```

Used RKE to deploy kubernetes, latest version: RKE 1.22
https://github.com/rancher/rke/releases/tag/v1.2.2

-----------------------------------------------------
Helm v3.3.4
```
[rke]$ helm version
version.BuildInfo{Version:"v3.3.4", GitCommit:"a61ce5633af99708171414353ed49547cf05013d", GitTreeState:"clean", GoVersion:"go1.14.9"}
```
-----------------------------------------------------

OS: All nodes running: Fedora release 32 (Thirty Two)
Docker version 19.03.13, build 4484c46d9d

-----------------------------------------------------
I have just added the helm repo and uninstalled, removed the namespace, then created a new namespace and installed as documented on helm repository. Install finish fine. But this script error persist.
Kubeconfig file has full admin permissions

**Comment by hugodopradofernandes** [comment]
This issue returned on v0.3.0

```
klustair-runnerjob
DEBUG:run Trivy on: quay.io/prometheus/prometheus:v2.21.0
klustair-runnerjob
DEBUG:run Trivy on: quay.io/prometheus/alertmanager:v0.21.0
klustair-runnerjob
DEBUG:run Trivy on: rancher/nginx-ingress-controller:nginx-0.35.0-rancher2
klustair-runnerjob
DEBUG:run Trivy on: klustair/klustair:v0.3.0
klustair-runnerjob
DEBUG:run Trivy on: weaveworks/scope:1.13.1
klustair-runnerjob
DEBUG:run Trivy on: klustair/klustair-frontend:v0.3.0-php-fpm
klustair-runnerjob
DEBUG:run Trivy on: quay.io/coreos/kube-state-metrics:v1.9.7
klustair-runnerjob
DEBUG:run Trivy on: squareup/ghostunnel:v1.5.2
klustair-runnerjob
DEBUG:run Trivy on: klustair/klustair-frontend:v0.3.0-apache
klustair-runnerjob
DEBUG:run Trivy on: docker.io/bitnami/wordpress:5.5.1-debian-10-r49
klustair-runnerjob
DEBUG:run Trivy on: kiwigrid/k8s-sidecar:0.1.209
klustair-runnerjob
INFO: Load Namespaces
klustair-runnerjob
INFO: Run Kubeaudit
klustair-runnerjob
not categorized: NamespaceHostNetworkTrue
klustair-runnerjob
not categorized: SeccompAnnotationMissing
klustair-runnerjob
not categorized: AutomountServiceAccountTokenTrueAndDefaultSA
klustair-runnerjob
not categorized: SeccompAnnotationMissing
klustair-runnerjob
not categorized: NamespaceHostNetworkTrue
klustair-runnerjob
not categorized: NamespaceHostPIDTrue
klustair-runnerjob
not categorized: SeccompAnnotationMissing
klustair-runnerjob
not categorized: SeccompAnnotationMissing
klustair-runnerjob
not categorized: AutomountServiceAccountTokenTrueAndDefaultSA
klustair-runnerjob
not categorized: NamespaceHostNetworkTrue
klustair-runnerjob
not categorized: NamespaceHostPIDTrue
klustair-runnerjob
not categorized: SeccompAnnotationMissing
klustair-runnerjob
INFO: Load Pod an Container informations
klustair-runnerjob
INFO: Start Trivy analysis
klustair-runnerjob
INFO: Load trivy Vulnerabilities
klustair-runnerjob
ERROR: could not parse quay.io/external_storage/nfs-client-provisioner:v3.1.0-k8s1.11
klustair-runnerjob
INFO: Save data to DB
klustair-runnerjob
INFO: Got a DB Connection.
klustair-runnerjob
Traceback (most recent call last):
klustair-runnerjob
File "./app/runner.py", line 343, in <module>
klustair-runnerjob
run()
klustair-runnerjob
File "./app/runner.py", line 285, in run
klustair-runnerjob
db.saveReport(report)
klustair-runnerjob
File "/app/database.py", line 71, in saveReport
klustair-runnerjob
self.__cursor.execute('''INSERT INTO k_reports(
klustair-runnerjob
psycopg2.errors.UndefinedTable: relation "k_reports" does not exist
klustair-runnerjob
LINE 1: INSERT INTO k_reports(
klustair-runnerjob
^
```

**Comment by mms-gianni** [comment]
This error could have caused this error : 
https://github.com/klustair/klustair-frontend/commit/94508d5de90d6ea720c8298aadb3dda11f60a1b9#diff-85e17f844b483d8f92e799870ff801eb3821e17e56b2c72b6da12ea51ba43778 


To check the onwership of the table run this SQL command on your Postgres: 
```
postgres=# select * from pg_tables where tablename like 'k_%';
 schemaname |       tablename        | tableowner | tablespace | hasindexes | hasrules | hastriggers | rowsecurity
------------+------------------------+------------+------------+------------+----------+-------------+-------------
 public     | k_namespaces           | postgres   |            | t          | f        | t           | f
 public     | k_audits               | postgres   |            | t          | f        | t           | f
 public     | k_pods                 | postgres   |            | t          | f        | t           | f
 public     | k_containers           | postgres   |            | t          | f        | t           | f
 public     | k_reports              | postgres   |            | t          | f        | t           | f
 public     | k_vulnsummary          | postgres   |            | t          | f        | t           | f
 public     | k_images               | postgres   |            | t          | f        | t           | f
 public     | k_vuln_anchore         | postgres   |            | t          | f        | t           | f
 public     | k_container_has_images | postgres   |            | t          | f        | t           | f
 public     | k_cwe                  | postgres   |            | t          | f        | f           | f
 public     | k_vuln_trivy           | postgres   |            | t          | f        | t           | f
 public     | k_config_runner        | postgres   |            | t          | f        | f           | f
 public     | k_reports_summaries    | postgres   |            | t          | f        | t           | f
 public     | k_vulnwhitelist        | postgres   |            | t          | f        | f           | f
 public     | k_target_trivy         | postgres   |            | t          | f        | t           | f
(15 rows)
```

Another try could be to set the env var DB_USERNAME to tableowner. 

But since version 3.0 has breaking changes in the Database anyway (introducing trivy targets), and all reports have to be deleted, you may want to reinstall the DB.

**Comment by hugodopradofernandes** [comment]
I will take a look at the table permission.
About the reinstall, I did reinstall all components, it's from a fresh install of v0.3.0

**Comment by mms-gianni** [comment]
I had several issues with migrations when i tried a fresh installation. But they are fixed now in Image tags >0.3.0 

https://github.com/klustair/klustair-frontend/commit/58a8426fd6772194f15462881d35cebe7e91270c 
https://github.com/klustair/klustair-frontend/commit/39687a9f1a540a1a23355d0108f9e48d9a1840d7 

But none of them resulted in in the your SQL Error.
